File: client/src/game/conf/AppConfig.ts
class AppConfig {
    private static _dic:Dictionary<AppConfig> = new Dictionary<AppConfig>();
    private static _arr:Array<AppConfig> = new Array<AppConfig>();

	 public Id :number ;
     public SocketIp :string ;
     public SocketPort :number ;
     public HttpIp :string ;
     public HttpPort :number ;
    

    
    public constructor(){

    }


    public static get Dic():Dictionary<AppConfig>{
        return  AppConfig._dic;
    }
    public static get Arr():Array<AppConfig>{
        return AppConfig._arr;
    }
    public static parse(json:any[]){
        if(json && json.length > 0){
			for(var i = 0; i < json.length; i++){
				let config:AppConfig = <AppConfig>json[i];
				AppConfig._arr.push(config);
				AppConfig._dic.add(config.Id,config);
			}
        }else{
            console.error("AppConfig loader fail!");
        }
    }
}

File: server/src/game/tools/conf/output/go/AppConfig.go
package conf

import (
	//"bufio"
	"encoding/json"
	"fmt"
	//"io"
	"io/ioutil"
	"os"
	//"strings"
)

type AppConfigElem struct {
	Id int
	SocketIp string
	SocketPort int
	HttpIp string
	HttpPort int
	
}

type AppConfig0 struct {
	Dic map[int]*AppConfigElem
	Arr []*AppConfigElem
}

var (
	AppConfig AppConfig0
)

func loadAppConfig(path string) {
	file, err := os.Open(path)
	defer func() {
		if file != nil {
			file.Close()
		}
	}()
	if err != nil {
		panic(fmt.Sprint("load config error, path=", path))
	}

	AppConfig.Dic = make(map[int]*AppConfigElem)
	AppConfig.Arr = make([]*AppConfigElem, 0, 32)

	bt,err  :=  ioutil.ReadAll(file)
	json.Unmarshal(bt, &AppConfig.Arr)
	for _,conf := range AppConfig.Arr{
		AppConfig.Dic[conf.Id] = conf
	}
	
}
